### PR TITLE
grafana/toolkit: Replace rollup with webpack

### PR DESCRIPTION
**Pull request by dprokop**
@ryantxu @torkelo this is an effort fo replacing Rollup with webpack for shared bundle config. 

Multiple entries handled the same way we did with Rollup (looking for module.ts(x) files and defining those as separate entries).

00a69191c0c8f5916e8bf2b5560450cd289c3256 puslished on npm as https://www.npmjs.com/package/@grafana/toolkit/v/6.3.0-alpha.20 (next channel)

TODO: 
- [x] Enable common chunks - @ryantxu is https://github.com/lapnap/grafana-live-app good for testing? (parked for now)
- [ ] Enable theme css extraction
  - [x] Add support for common conventions (see README & https://github.com/grafana/grafana/issues/17819)
  - [ ] Verify SASS setup
  - [ ] Verify static assets support
  - [ ] Add PostCSS loader

### CODE REVIEW

**Comment by ryantxu** [comment]
This is looking much better than rollup.  I'll keep a log here of issues that we need to solve:

[Influx Flux](https://github.com/grafana/influxdb-flux-datasource/pull/41)
Type issue in jest?
image

**Comment by dprokop** [comment]
@ryantxu thanks a lot for testing this out!

File: packages/grafana-toolkit/src/cli/tasks/plugin.build.ts [hunk 1]
```diff
@@ -14,6 +14,19 @@ interface PrecommitOptions {}
 // @ts-ignore
 export const clean = useSpinner<void>('Cleaning', async () => await execa('rimraf', ['./dist']));
 
+export const prepare = useSpinner<void>('Prepareing', async () => {
+  // Make sure a local tsconfig exists.  Otherwise this will work, but have odd behavior
```
Comment (dprokop) [inline packages/grafana-toolkit/src/cli/tasks/plugin.build.ts]: Cool, this is what react-scrips do as well

**Comment by dprokop** [comment]
@ryantxu New Relic DS issues fixed with 52c797b01ba1dc18f0c242d4235fff8eef4a5df8 . According to Jest docs [`roots`](https://jestjs.io/docs/en/configuration#roots-array-string) option is used to "locate test and source files". I have removed that from the shared config.

**Comment by dprokop** [comment]
@ryantxu https://github.com/lapnap/grafana-live-app surfaced two problems:
1. absolute imports, i.e. `import XXX from 'app/...'`, `import XXX from 'features/...'`. This is solved now with webpack config that allows relative imports from src, see https://github.com/grafana/grafana/pull/17850/commits/21c4a7e8f5204b63427eb1f2adb02178a41f5368 for details
2. absolute imports from `... from 'app/....'`. Since we are externalising `app/*` imports (https://github.com/grafana/grafana/blob/21c4a7e8f5204b63427eb1f2adb02178a41f5368/packages/grafana-toolkit/src/config/webpack.plugin.config.ts#L96), the internal live app imports were externalised as well, causing the app fail to load in grafana. I have changed the absolute imports to relative ones to fix that. Will open PR for that in grafana-live-app when u add me to that repo :)

**Comment by dprokop** [comment]
Short update on common chunks extraction, spent quite a lot of time on this today. Issue is that (most probably) the common chunks files need to be loaded before before the actual plugin's module. Not sure yet what's the best way to do that yet, seems like sth we can park for a while and focus on next steps.

File: packages/grafana-toolkit/package.json [hunk 2]
```diff
@@ -42,7 +42,9 @@
     "inquirer": "^6.3.1",
     "jest": "24.8.0",
     "jest-cli": "^24.8.0",
+    "jest-coverage-badges": "^1.1.2",
     "lodash": "4.17.11",
+    "node-sass": "^4.12.0",
```
Comment (dprokop) [inline packages/grafana-toolkit/package.json]: 👍

**Comment by ryantxu** [comment]
@dprokop -- I think the best app to test with is [kentik-app](https://github.com/grafana/kentik-app/pull/78)

This has two key webpack issues:
1. multiple entry points
2. light/dark scss

**Review by ryantxu — APPROVED** [review]
This is looking great -- I think we should merge soon and add improvements in additional PRs

**Pull request merged**